### PR TITLE
[Transaction] Call getCause() after exception wrapped by completeFuture

### PR DESCRIPTION
**Pull request by liangyepianzhou**
Master Issue https://github.com/apache/pulsar/issues/13918
### Motivation & Modification
Call getCause() after exception wrapped by completeFuture
*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionCoordinatorClientImpl.java [hunk 1]
```diff
@@ -168,7 +168,7 @@ public TxnID newTransaction(long timeout, TimeUnit unit) throws TransactionCoord
         try {
             return newTransactionAsync(timeout, unit).get();
         } catch (Exception e) {
-            throw TransactionCoordinatorClientException.unwrap(e);
+            throw TransactionCoordinatorClientException.unwrap(e.getCause());
```
Comment (Jason918) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionCoordinatorClientImpl.java:171]: Better to check CompletionException in `TransactionCoordinatorClientException.unwrap` ?

Comment (liangyepianzhou) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionCoordinatorClientImpl.java:171]: Good idea! I will modify it immediately

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/transaction/TransactionCoordinatorClientException.java [hunk 2]
```diff
@@ -119,18 +119,13 @@ public static TransactionCoordinatorClientException unwrap(Throwable t) {
         } else if (t instanceof InterruptedException) {
             Thread.currentThread().interrupt();
             return new TransactionCoordinatorClientException(t);
-        }  else if (!(t instanceof ExecutionException)) {
+        }  else if (t instanceof CoordinatorNotFoundException) {
+            return (CoordinatorNotFoundException) t;
+        } else if (t instanceof InvalidTxnStatusException) {
+            return (InvalidTxnStatusException) t;
+        } else if (t instanceof ExecutionException) {
```
Comment (Jason918) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/transaction/TransactionCoordinatorClientException.java:126]: It's better to check `CompletionException` together.

**Review by Jason918 — DISMISSED** [review]

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/transaction/TransactionCoordinatorClientException.java [hunk 3]
```diff
@@ -119,18 +120,13 @@ public static TransactionCoordinatorClientException unwrap(Throwable t) {
         } else if (t instanceof InterruptedException) {
             Thread.currentThread().interrupt();
             return new TransactionCoordinatorClientException(t);
-        }  else if (!(t instanceof ExecutionException)) {
+        }  else if (t instanceof CoordinatorNotFoundException) {
```
Comment (codelipenghui) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/transaction/TransactionCoordinatorClientException.java:123]:
```suggestion
        } else if (t instanceof CoordinatorNotFoundException) {
```

**Review by codelipenghui — COMMENTED** [review]
The change looks good to me, could you please add tests for the new changes?

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**